### PR TITLE
Mumax3c support

### PR DESCRIPTION
**Pull request by samjrholt**

### CODE REVIEW

**Comment by samjrholt** [comment]
@lang-m I've made `micromagneticdata` work with `mumax3c` please can you take a look as you might know a better way to do this.

**Comment by samjrholt** [comment]
If this option is acceptable I will change the documentation for the new `mx3` function.

**Comment by lang-m** [comment]
Documentation needs some polishing and we need a few small pre-computed `mumax3` examples.

File: micromagneticdata/drive.py [hunk 1]
```diff
@@ -60,49 +60,32 @@ class Drive(md.AbstractDrive):
 
     """
 
+    def __new__(cls, name, number, dirname=".", x=None):
```
Comment (marijanbeg) [inline micromagneticdata/drive.py:63]: Nice :)

File: micromagneticdata/abstract_drive.py [hunk 2]
```diff
@@ -103,7 +103,7 @@ def m0(self):
         Field(...)
 
         """
-        return df.Field.fromfile(self._m0_path)
+        return df.Field.fromfile(str(self._m0_path))
```
Comment (marijanbeg) [inline micromagneticdata/abstract_drive.py:106]: Is this because `Field.fromfile` does not accept `Path` type?

File: micromagneticdata/drive.py [hunk 3]
```diff
@@ -130,11 +113,12 @@ def info(self):
...
 
         """
-        with open(os.path.join(self.path, "info.json")) as f:
+        with (self.drive_path / "info.json").open() as f:
             return json.load(f)
 
     @property
-    def mif(self):
+    @abc.abstractmethod
+    def input_script(self):
```
Comment (marijanbeg) [inline micromagneticdata/drive.py:121]: Maybe `caclulator_script` would be a better name.

File: micromagneticdata/drive.py [hunk 4]
```diff
@@ -212,7 +164,7 @@ def ovf2vtk(self, dirname=None):
 
         """
         if dirname is None:
-            dirname = self.path
+            dirname = str(self.drive_path)
         for i, filename in enumerate(self._step_files):
             vtkfilename = "drive-{}-{:07d}.vtk".format(self.number, i)
             df.Field.fromfile(filename).write(os.path.join(dirname, vtkfilename))
```
Comment (marijanbeg) [inline micromagneticdata/drive.py:170]: `os.path.join` -> `pathlib.Path() / ...`

File: micromagneticdata/drive.py [hunk 5]
```diff
@@ -212,7 +164,7 @@ def ovf2vtk(self, dirname=None):
 
         """
         if dirname is None:
-            dirname = self.path
+            dirname = str(self.drive_path)
```
Comment (marijanbeg) [inline micromagneticdata/drive.py:167]: `dirname = dirname or str(self.drive_path)` short-circuit

File: micromagneticdata/drive.py [hunk 6]
```diff
@@ -212,7 +164,7 @@ def ovf2vtk(self, dirname=None):
...
         for i, filename in enumerate(self._step_files):
             vtkfilename = "drive-{}-{:07d}.vtk".format(self.number, i)
```
Comment (marijanbeg) [inline micromagneticdata/drive.py:169]: f-string?

**Review by marijanbeg — APPROVED** [review]
Very nice!

File: micromagneticdata/abstract_drive.py (hunk 2 above)
Comment (lang-m) [reply micromagneticdata/abstract_drive.py:106]: Yes. I would suggest to keep it for now and switch to `pathlib.Path` everywhere (in all packages) separately.

Comment (marijanbeg) [reply micromagneticdata/abstract_drive.py:95]: +1

**Pull request merged**